docs/pipeline_configuration_files.md

```diff
@@ -0,0 +1,96 @@
+`status` and `name`, which are configurations for the pipeline itself.
+
+To create connectors in that pipeline, simply add another map under the pipeline map, and call it `connectors`.
+
+To create processors, either add a `processors` map under a pipeline ID, or under a connector ID, depending on its parent.
+check this YAML file example with explanation for each field:
+
+``` yaml
+version: 1.0                    # parser version, the only supported version for now is 1.0 [mandatory]
+
+pipelines:                      # a map of pipelines IDs and their configurations.
+  pipeline1:                    # pipeline ID, has to be unique.
+    status: running             # pipelines status at startup, either running or stopped. [mandatory]
+    name: pipeline1             # pipeline name, if not specified, pipeline ID will be used as name. [optional]
+    describtion: desc           # pipeline describtion. [optional]
+    connectors:                 # a map of connectors IDs and their configurations.
+      con1:                     # connector ID, has to be unique per pipeline.
+        type: source            # connector type, either "source" or "destination". [mandatory]
+        plugin: builtin:file    # connector plugin. [mandatory]
+        name: con3              # connector name, if not specified, connector ID will be used as name. [optional]
+        settings:               # map of configurations keys and their values.
+          path: ./file1.txt     # for this example, the plugin "bultin:file" has only one configuration, which is path.
+      con2:
+        type: destination
+        plugin: builtin:file
+        name: file-dest
+        settings:
+          path: ./file2.txt
+        processors:             # a map of processor IDs and their configurations, "con2" is the processor parent.
+          proc1:                # processor ID, has to be unique for each parent
+            type: js            # processor type. [mandatory]
+            settings:           # map of processor configurations and values
+              Prop1: string
+    processors:                 # processor IDs, that have the pipeline "pipeline1" as a parent.
+      proc2: 
+        type: js
+        settings:
+          prop1: ${ENV_VAR}     # yon can use environmental variables by wrapping them in a dollar sign and curly braces ${}.
+```
+
+If the file is invalid (missed a mandatory field, or has an invalid configuration value), then the pipeline that has the
+invalid value will be skipped, with an error message logged.
+
+If two pipelines in one file have the same ID, or the `version` field was not specified, then the file would be 
+non-parsable and will be skipped with an error message logged.
+
+If two pipelines from different files have the same ID, the second pipeline will be skipped, with an error message
+specifying which pipeline was not provisioned.
```

Comment:
because this will be discovered while provisioning, the first case is an invalid yaml file which will be known while parsing, and we parse and provision one file at a time. So we wouldn't know if another file has the same pipelineID, and if it does then we only want to skip the pipeline not skip the whole file like in the first case.

Reply:
> if it does then we only want to skip the pipeline not skip the whole file like in the first case.

This clears it up for me, thanks!